Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml

```diff
@@ -69,36 +69,58 @@
   id: Musket
   description: This should've been in a museum long before you were born. Uses .60 anti-materiel ammo.
   components:
+  - type: Sharp
   - type: Item
     size: Large
   - type: Sprite
     sprite: Objects/Weapons/Guns/Snipers/musket.rsi
+    state: base
   - type: Clothing
     sprite: Objects/Weapons/Guns/Snipers/musket.rsi
   - type: Gun
+    useKey: false
     selectedMode: SemiAuto
     availableModes:
     - SemiAuto
+  - type: UseDelayOnShoot
+  - type: UseDelay
+    delay: 8 #it's a musket
   - type: BallisticAmmoProvider
     whitelist:
       tags:
       - CartridgeAntiMateriel
     capacity: 1
     proto: CartridgeAntiMateriel
-
+  - type: MeleeWeapon
+    wideAnimationRotation: -135
+    damage:
+      types:
+        Piercing: 15 #you fucking stab em
+        Bloodloss: 2 #no way to apply bleed, triangular bayonet wounds are hard to fix(source:that one copypasta)
+    angle: 0
+    animation: WeaponArcThrust
+    soundHit:
+      path: /Audio/Weapons/bladeslice.ogg
+    
 - type: entity
   name: flintlock pistol
   parent: BaseWeaponSniper
   id: WeaponPistolFlintlock
   description: A pirate's companion. Yarrr! Uses .60 anti-materiel ammo.
   components:
+  - type: Gun
+    minAngle: 30
```

Comment:
Does this make it always shoot off 30 degrees left or right?

Reply:
nope, just a range

Reply:
I think
Fuck.